go.mod

```diff
@@ -2,11 +2,14 @@ module github.com/jackc/puddle/v2
 
 go 1.18
 
-require github.com/stretchr/testify v1.8.0
+require (
+	github.com/stretchr/testify v1.8.0
+	go.uber.org/atomic v1.10.0
```

Comment:
Given that you use only a single type and two methods on it (Add and Load), I'd suggest just copying the relevant type and methods here, particularly since it'll be obviated eventually by sync/atomic. [A little copying is better than a little dependency.](https://go-proverbs.github.io)

Reply:
This package is pretty standard and well supported. It's not some obscure package with 10 stars. In this case dependency feels OK to me, but this is definitely possible if @jackc will not want the dependency.

Reply:
I'll obviously defer to @jackc on this, but I feel pretty strongly that the benefits of copy/pasting less than a dozen lines of trivial code far exceed the costs.

Reply:
Keep in mind that this will be there for 4 months and this library is pretty stable for last few years. I understand the argument with dependencies and we all remember `leftpad`, but I thing that importing well-managed leaf (doesn't import other libs) library is by far not problematic.

Reply:
Resolved now - we will use stdlib atomic types.